docs/rule/no-route-action.md

```diff
@@ -0,0 +1,51 @@
+# no-route-action
+
+This rule disallows the usage of `route-action`.
+
+[ember-route-action-helper](https://github.com/DockYard/ember-route-action-helper) is a popular addon that it's only purpose was to avoid creating controllers, something we no longer have a desire to do given the current state of the framework.
+
+Most route actions should either be sent to the controller first or encapsulated within a downstream component instead. We should never be escaping the DDAU hierarchy to lob actions up to the route.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+{{custom-component onUpdate=(route-action 'updateFoo')}}
+```
+
+```hbs
+{{custom-component onUpdate=(route-action 'updateFoo' 'bar')}}
+```
+
+```hbs
+<CustomComponent onUpdate={{route-action 'foo'}} />
```

Comment:
```suggestion
<CustomComponent @onUpdate={{route-action 'foo'}} />
```

Reply:
This is the case for every `<CustomComponent />` occurrence here and in the test file.